proof/refine/ARM/KHeap_R.thy

```diff
@@ -24,6 +24,37 @@ abbreviation (input)
 where
   "set_obj' ptr obj s \<equiv> set_ko' ptr (injectKO obj) s"
 
+(* FIXME RT: bikeshedding *)
```

Comment:
To start off the bikeshedding, can you say a bit about the use case where you found this helped? Was it mainly that you didn't want to deal with `is_aligned` and `ps_clear`?

Reply:
This started with @mktnk3 pointing out that in lots of places, `ksPSpace s p = Some blah` is a weaker assumption _and_ weaker conclusion for various properties (I believe `sym_refs`, in my case). Not sure the use-case really affects the bikeshedding, though.

Reply:
Generally speaking, I am quite hesitant of introducing all these. I think this is something that is not supposed to be introduced in bits like this. If we are to take that approach, maybe doing it more fully and generally would be better. Otherwise we run the risk of diverging. Also, I am rather reluctant in introducing abbreviations for such basic one-liners. To me, these will potentially hide information, and that could potentially mean extra load on everyone’s mind. But this is also just my preference as well. @lsf37 can I ask for your option on this?

Reply:
And I also should say that, at this point, I myself don’t have a clear picture of where my comments you are referring to was heading to, if it ever does.

Reply:
Pinging @lsf37 again on this. I think I have been using projections instead in most of the situations when I want something like this. For example, `ks_reply_at’ r p s` here is equivalent to `replies_of’ s p = Some r`. So, I am not convinced that we should be making a step in the direction suggested here at this point.

Reply:
Alright, I think I'll rip this out for now, maybe switch to projections when appropriate.

Reply:
Ha, wow, this was almost entirely unused. Oh well.

Reply:
Meta comment: When selecting code to make a comment on a pull request, it would be good to use a multi-line selection for the comment, selecting as many lines as might be relevant to the discussion that could follow from the initial comment.
In the conversation UI, GitHub never shows context *below* the lines that were selected, despite a [7-year-old feature request](https://github.com/isaacs/github/issues/97). Also, GitHub doesn't give a way to get more context in the conversation UI, so one has to go digging in the diff for the file to find the context. And if the branch has been force-pushed, that context can be extremely difficult to find, if it can be found at all.
Explicitly selecting a more generous amount of context would therefore help people like me who are late to the party and want to try to catch up on all this great PR communication. :-)

Reply:
> Meta comment: When selecting code to make a comment on a pull request, it would be good to use a multi-line selection for the comment, selecting as many lines as might be relevant to the discussion that could follow from the initial comment.

Right. that makes sense. I didn’t know that that was possible, but it should be, I suppose!